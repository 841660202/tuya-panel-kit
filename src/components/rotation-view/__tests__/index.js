import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import RotationView from '../index';

describe('RotationView Component', () => {
  it('basic render', () => {
    const component = renderer.create(<RotationView active={true} />);
    expect(component.toJSON()).toMatchSnapshot();

    const component2 = renderer.create(<RotationView active={false} />);
    expect(component2.toJSON()).toMatchSnapshot();
  });

  it('should update properly', () => {
    const component = renderer.create(<RotationView active={true} key={'some-key'}/>);
    const instance = component.root.instance;
    instance.startAnimation = jest.fn();
    instance.stopAnimation = jest.fn();
    jest.useFakeTimers()
    jest.runAllTimers();

    component.update(<RotationView active={false} key={'some-key'}/>);
    expect(instance.stopAnimation).toHaveBeenCalled();

    component.update(<RotationView active={true} key={'some-key'}/>);
    expect(instance.startAnimation).toHaveBeenCalled();

    instance.componentWillUnmount();
    expect(instance.stopAnimation).toHaveBeenCalled();
  });


});
