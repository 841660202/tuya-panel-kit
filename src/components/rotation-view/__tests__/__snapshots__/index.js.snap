// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RotationView Component basic render 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "transform": Array [
        Object {
          "rotate": "0deg",
        },
      ],
    }
  }
/>
`;

exports[`RotationView Component basic render 2`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "transform": Array [
        Object {
          "rotate": "0deg",
        },
      ],
    }
  }
/>
`;
